common/persistence/execution_manager.go

```diff
@@ -121,6 +120,38 @@ func (m *executionManagerImpl) UpdateWorkflowExecution(
 	request *UpdateWorkflowExecutionRequest,
 ) (*UpdateWorkflowExecutionResponse, error) {
 
+	var updateWorkflowNewEvents []*InternalAppendHistoryNodesRequest
+	var newWorkflowNewEvents []*InternalAppendHistoryNodesRequest
+	var updateWorkflowHistoryDiff, newWorkflowHistoryDiff int64 = 0, 0
+	for _, workflowEvents := range request.UpdateWorkflowEvents {
+		newEvents, err := m.serializeWorkflowEvents(request.ShardID, workflowEvents)
+		newEvents.ShardID = request.ShardID
+		if err != nil {
+			return nil, err
+		}
+		if newEvents != nil {
+			updateWorkflowNewEvents = append(updateWorkflowNewEvents, newEvents)
+			updateWorkflowHistoryDiff += int64(len(newEvents.Node.Events.Data))
+		}
+	}
+	if updateWorkflowHistoryDiff > 0 {
+		request.UpdateWorkflowMutation.ExecutionInfo.ExecutionStats.HistorySize += updateWorkflowHistoryDiff
+	}
+
+	for _, workflowEvents := range request.NewWorkflowEvents {
+		newEvents, err := m.serializeWorkflowEvents(request.ShardID, workflowEvents)
+		if err != nil {
+			return nil, err
+		}
+		if newEvents != nil {
+			newWorkflowNewEvents = append(newWorkflowNewEvents, newEvents)
+			newWorkflowHistoryDiff += int64(len(newEvents.Node.Events.Data))
+		}
+	}
+	if newWorkflowHistoryDiff > 0 {
```

Comment:
same here

Reply:
NewWorkflowSnapshot could be nil.

Reply:
change to explicit check nil on NewWorkflowSnapshot.